spyder/utils/programs.py

```diff
@@ -710,7 +712,25 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                 else:
                     run_program(program['cmd'], arglist)
                 return
-        # TODO: Add a fallback to OSX
+    elif sys.platform == 'darwin':
+        f = tempfile.NamedTemporaryFile('wt', prefix='run_spyder_',
+                                        suffix='.sh', dir=get_temp_dir(),
+                                        delete=False)
+        if wdir:
+            f.write('cd {}\n'.format(wdir))

```

Comment:
Great work @steff456!  

Lets add a test to check for spaces in paths

```
wdir = '/users/tefa/Dekstop/carpeta con espacio/'
wdir = '/users/tefa/Dekstop/carpeta con espacio/archivo con espacio'
# It may fail here
'cd {}'.format(wdir)
```

Reply:
I should create a new one or modify the existing one?

Reply:
Lets try to create a parametrized test?

like done here https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/editor/widgets/tests/test_goto.py#L31 ?